docs/advanced-ai/examples/understand-vector-databases.md

```diff
@@ -0,0 +1,51 @@
+---
+title: What are vector databases?
+description: Understand vector databases. Learn how n8n provides vector databases, along with the key components to work with them, including embeddings, retrievers, and document loaders.
+contentType: explanation
+---
+
+# What are vector databases?
+
+Vector databases store information as numbers:
+
+> A vector database is a type of database that stores data as high-dimensional vectors, which are mathematical representations of features or attributes. ([source](https://learn.microsoft.com/en-us/semantic-kernel/memories/vector-db){:target=_blank .external-link})
+
+This enables fast and accurate similarity searches. With a vector database, instead of using conventional database queries, you can search for relevant data based on semantic and contextual meaning.
+
+## A very simplified example
+
+A vector database could store the sentence "n8n is a source-available automation tool that you can self-host", but instead of storing it as text, the vector database stores numbers that represent its features. This doesn't mean turning each letter in the sentence into a number. Instead, the vectors in the vector database describe the sentence. 
```

Comment:
This might be the case for label embedding but not quite for "text relatedness" embeddings we use. These are not simple one-hot encodings where each dimension corresponds to a single semantic feature (like "automation tool", "source available", etc). Instead, each dimension in the embedding vector captures some complex combination of semantic features, and it's not usually possible to interpret the individual dimensions in isolation.

For example, a real embedding for the sentence "n8n is a source-available automation tool that you can self-host" would be something like:
```
[0.23, -0.12, 0.78, -0.55, 0.92, ..., 0.14]
```

Where the vector might have hundreds or even thousands of dimensions. Each dimension doesn't correspond to a single interpretable feature, but rather the pattern across all the dimensions encodes the overall semantics of the sentence. Similarly, the embeddings for related sentences like "Zapier is an automation tool" would not simply be subsets of the dimensions from the n8n sentence. Instead, they would be entirely separate high-dimensional vectors:
```
[0.19, -0.08, 0.81, -0.60, 0.87, ..., 0.22]
```
The semantic similarity between the sentences would be captured by the geometric proximity of these high-dimensional vectors, not by overlapping individual dimensions.